pkg/sysinfo/sysinfo.go

```diff
@@ -102,6 +117,128 @@ type cgroupPids struct {
 	PidsLimit bool
 }
 
+// ValidateHugetlb check whether hugetlb pagesize and limit legal
+func (c cgroupHugetlbInfo) ValidateHugetlb(pageSize string, limit uint64) (string, []string, error) {
+	var (
+		w   []string
+		err error
+	)
+	if pageSize != "" {
+		if err = isHugepageSizeValid(pageSize); err != nil {
+			return "", w, err
+		}
+	} else {
+		pageSize, err = c.GetDefaultHugepageSize()
+		if err != nil {
+			return "", w, fmt.Errorf("Failed to get system hugepage size")
+		}
+	}
+
+	warning, err := isHugeLimitValid(pageSize, limit)
+	w = append(w, warning...)
+	if err != nil {
+		return "", w, err
+	}
+
+	return pageSize, w, nil
+}
+
+// isHugeLimitValid check whether input hugetlb limit legal
+// it will check whether the limit size is times of size
+func isHugeLimitValid(size string, limit uint64) ([]string, error) {
+	var w []string
+	sizeInt, err := units.RAMInBytes(size)
```

Comment:
I have to agree with @justincormack that we may be doing too much validation. For example, we now not only validate if `size` is a valid size (i.e. it's in the list of configured sizes on the host), but also if we're able to parse/interpret that size to something useful.

Perhaps just validate:

- if the specified size is found on the machine (`isHugepageSizeValid()`) (or not even, but that can be discussed)
- if the specified limit is valid (i.e., we're able to parse the size, in bytes)

And leave the rest to the runtime/kernel.

Reply:
```
$echo 5242880 > hugetlb.2MB.limit_in_bytes
-bash: echo: write error: Invalid argument
```

I dont think this error message is kind of confused, so just validate limit here.